[14.0][ADD] shopfloor_reception_purchase_partner_ref

mt-software-de opened the pull request.

mt-software-de committed:
[ADD] shopfloor_reception_purchase_partner_ref

Show the partner_ref field of a purchase order on the reception scenario

simahawk commented on shopfloor_reception_purchase_partner_ref/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,2 @@
+Shopfloor extensio of the reception scenario.
```
```suggestion
Shopfloor extension of the reception scenario.
```

simahawk commented on shopfloor_reception_purchase_partner_ref_mobile/static/src/scenario/reception.js:
```diff
@@ -0,0 +1,29 @@
+
+import {process_registry} from "/shopfloor_mobile_base/static/wms/src/services/process_registry.js";
+
+const reception_scenario = process_registry.get("reception");
```
Can you please try a std Vue.js override? 
```js
Vue.component("reception", {
    "methods": ...
})
``` 

Then you should be able to use `this.$super(reception_scenario).manual_select_options_for_select_document()` to get the original result.

mt-software-de replied:
This didn't worked. 
How should it work? Because it is a child of shopfloor_mobile_base/static/wms/src/services/registry.js shopfloor_mobile_base/static/wms/src/services/process_registry.js

mt-software-de replied:
Or did i miss something?

jbaudoux replied:
@simahawk can you answer?

mt-software-de replied:
I will resolve this, because @simahawk answered it already in an other PR. 
https://github.com/OCA/wms/pull/647#discussion_r1274454764

simahawk commented on shopfloor_reception_purchase_partner_ref/actions/schema.py:
```diff
@@ -0,0 +1,19 @@
+
+    def picking(self):
+        res = super().picking()
+        res.update({"purchase_order": self._schema_dict_of(self.purchase_order())})
```
This will be used by any picking parser.
I'd move move these changes to a base module for purchase (eg: `shopfloor_purchase_base`) that can be reused  and here add a conditional argument like `with_purchase_order` to be used only when needed.
WDYT?

mt-software-de replied:
with_purchase_order would be great. 
But then we should also change the property _picking_parser and the schema for picking. 
![image](https://github.com/OCA/wms/assets/78783475/23c0ce13-f854-4380-8683-df8bda1150ff)
![image](https://github.com/OCA/wms/assets/78783475/47b7b1b7-7930-4f80-a9f5-b8768e623aa2)

and also we have to change the shopfloor_reception to call of those methods with "with_purchase_order"
![image](https://github.com/OCA/wms/assets/78783475/ff68e0a7-37dd-4394-928d-af6516a93ed0)

I think it would be much easier, then to always add the purchase order to the picking.

mt-software-de replied:
I changed this now again. 
I added the kw with_purchase_order

simahawk commented:
One more thing: I'd recommend to move the the schange on the `shopfloor_reception_mobile` to another PR to speed its merge and to be able to bump  version on merge.

mt-software-de committed:
[ADD] shopfloor_purchase_base: Purchase order data on picking

Add optional purchase order data to picking data
To get the purchase order within the picking data
call data.picking with kwarg with_purchase_order

mt-software-de commented:
@jbaudoux could you please review this PR?

mt-software-de committed:
[IMP] shopfloor_reception: adding kwargs to _data_for_stock_picking

mt-software-de committed:
[IMP] shopfloor: data picking parser into own method

simahawk reviewed (commented):
LG.
@mt-software-de can you please shorten commit msgs and keep descriptions into the commit description?

mt-software-de commented:
> LG. @mt-software-de can you please shorten commit msgs and keep descriptions into the commit description?

Done

simahawk commented:
@mt-software-de the last commit is still too long but in fact you don't need any additional info beside `[add] $module_name` :wink:

jbaudoux commented:
/ocabot merge minor
